// @cds/angular/esm2020/internal-components/close-button/internal-components-close-button.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/internal-components/close-button/register.js';
import { CdsInternalCloseButtonDirective } from './cds-internal-close-button.directive';
import * as i0 from "@angular/core";
export class CdsInternalComponentsCloseButtonModule {
}
CdsInternalComponentsCloseButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsCloseButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInternalComponentsCloseButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsCloseButtonModule, declarations: [CdsInternalCloseButtonDirective], imports: [CommonModule], exports: [CdsInternalCloseButtonDirective] });
CdsInternalComponentsCloseButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsCloseButtonModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsCloseButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInternalCloseButtonDirective,
                    ],
                    exports: [
                        CdsInternalCloseButtonDirective,
                    ],
                }]
        }] });
