// @cds/angular/esm2020/divider/divider.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/divider/register.js';
import { CdsDividerDirective } from './cds-divider.directive';
import * as i0 from "@angular/core";
export class CdsDividerModule {
}
CdsDividerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDividerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsDividerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDividerModule, declarations: [CdsDividerDirective], imports: [CommonModule], exports: [CdsDividerDirective] });
CdsDividerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDividerModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDividerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsDividerDirective,
                    ],
                    exports: [
                        CdsDividerDirective,
                    ],
                }]
        }] });
