// @cds/angular/esm2020/selection-panels/checkbox/selection-panels-checkbox.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/selection-panels/checkbox/register.js';
import { CdsCheckboxPanelDirective } from './cds-checkbox-panel.directive';
import * as i0 from "@angular/core";
export class CdsSelectionPanelsCheckboxModule {
}
CdsSelectionPanelsCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsSelectionPanelsCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsCheckboxModule, declarations: [CdsCheckboxPanelDirective], imports: [CommonModule], exports: [CdsCheckboxPanelDirective] });
CdsSelectionPanelsCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsCheckboxModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsCheckboxPanelDirective,
                    ],
                    exports: [
                        CdsCheckboxPanelDirective,
                    ],
                }]
        }] });
