// @cds/angular/esm2020/signpost/signpost.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/signpost/register.js';
import { CdsSignpostDirective } from './cds-signpost.directive';
import * as i0 from "@angular/core";
export class CdsSignpostModule {
}
CdsSignpostModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSignpostModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsSignpostModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSignpostModule, declarations: [CdsSignpostDirective], imports: [CommonModule], exports: [CdsSignpostDirective] });
CdsSignpostModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSignpostModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSignpostModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsSignpostDirective,
                    ],
                    exports: [
                        CdsSignpostDirective,
                    ],
                }]
        }] });
