// @cds/angular/esm2020/internal-components/overlay/internal-components-overlay.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/internal-components/overlay/register.js';
import { CdsInternalOverlayDirective } from './cds-internal-overlay.directive';
import * as i0 from "@angular/core";
export class CdsInternalComponentsOverlayModule {
}
CdsInternalComponentsOverlayModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsOverlayModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInternalComponentsOverlayModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsOverlayModule, declarations: [CdsInternalOverlayDirective], imports: [CommonModule], exports: [CdsInternalOverlayDirective] });
CdsInternalComponentsOverlayModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsOverlayModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsOverlayModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInternalOverlayDirective,
                    ],
                    exports: [
                        CdsInternalOverlayDirective,
                    ],
                }]
        }] });
