// @cds/angular/esm2020/range/range.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/range/register.js';
import { CdsRangeDirective } from './cds-range.directive';
import * as i0 from "@angular/core";
export class CdsRangeModule {
}
CdsRangeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRangeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsRangeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRangeModule, declarations: [CdsRangeDirective], imports: [CommonModule], exports: [CdsRangeDirective] });
CdsRangeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRangeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRangeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsRangeDirective,
                    ],
                    exports: [
                        CdsRangeDirective,
                    ],
                }]
        }] });
