// @cds/angular/esm2020/navigation/cds-navigation.directive.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
export class CdsNavigationDirective {
    constructor(elementRef) {
        this.cdsMotionChange = new EventEmitter();
        this.expandedChange = new EventEmitter();
        this.element = elementRef.nativeElement;
    }
    get expandedRoot() {
        return this.element.expandedRoot;
    }
    set expandedRoot(value) {
        this.element.expandedRoot = value === '' ? true : value;
    }
    ;
    get cdsMotion() {
        return this.element.cdsMotion;
    }
    set cdsMotion(value) {
        this.element.cdsMotion = value;
    }
    ;
    get expanded() {
        return this.element.expanded;
    }
    set expanded(value) {
        this.element.expanded = value === '' ? true : value;
    }
    ;
    get i18n() {
        return this.element.i18n;
    }
    set i18n(value) {
        this.element.i18n = value;
    }
    ;
    get onStartItemSlotChange() {
        return this.element.onStartItemSlotChange;
    }
    set onStartItemSlotChange(value) {
        this.element.onStartItemSlotChange = value;
    }
    ;
    get onItemSlotChange() {
        return this.element.onItemSlotChange;
    }
    set onItemSlotChange(value) {
        this.element.onItemSlotChange = value;
    }
    ;
    get updateChildrenProps() {
        return this.element.updateChildrenProps;
    }
    set updateChildrenProps(value) {
        this.element.updateChildrenProps = value;
    }
    ;
}
CdsNavigationDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
CdsNavigationDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: CdsNavigationDirective, selector: "cds-navigation", inputs: { expandedRoot: "expandedRoot", cdsMotion: "cdsMotion", expanded: "expanded", i18n: "i18n", onStartItemSlotChange: "onStartItemSlotChange", onItemSlotChange: "onItemSlotChange", updateChildrenProps: "updateChildrenProps" }, outputs: { cdsMotionChange: "cdsMotionChange", expandedChange: "expandedChange" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'cds-navigation' }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { expandedRoot: [{
                type: Input
            }], cdsMotion: [{
                type: Input
            }], expanded: [{
                type: Input
            }], i18n: [{
                type: Input
            }], onStartItemSlotChange: [{
                type: Input
            }], onItemSlotChange: [{
                type: Input
            }], updateChildrenProps: [{
                type: Input
            }], cdsMotionChange: [{
                type: Output
            }], expandedChange: [{
                type: Output
            }] } });
