// @cds/angular/esm2020/navigation/cds-navigation-item.directive.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class CdsNavigationItemDirective {
    constructor(elementRef) {
        this.element = elementRef.nativeElement;
    }
    get i18n() {
        return this.element.i18n;
    }
    set i18n(value) {
        this.element.i18n = value;
    }
    ;
    get active() {
        return this.element.active;
    }
    set active(value) {
        this.element.active = value === '' ? true : value;
    }
    ;
    get disabled() {
        return this.element.disabled;
    }
    set disabled(value) {
        this.element.disabled = value === '' ? true : value;
    }
    ;
    get expanded() {
        return this.element.expanded;
    }
    set expanded(value) {
        this.element.expanded = value === '' ? true : value;
    }
    ;
    get groupItem() {
        return this.element.groupItem;
    }
    set groupItem(value) {
        this.element.groupItem = value === '' ? true : value;
    }
    ;
    get hasFocus() {
        return this.element.hasFocus;
    }
    set hasFocus(value) {
        this.element.hasFocus = value;
    }
    ;
    get focusElement() {
        return this.element.focusElement;
    }
    set focusElement(value) {
        this.element.focusElement = value;
    }
    ;
    get itemText() {
        return this.element.itemText;
    }
    set itemText(value) {
        this.element.itemText = value;
    }
    ;
}
CdsNavigationItemDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationItemDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
CdsNavigationItemDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: CdsNavigationItemDirective, selector: "cds-navigation-item", inputs: { i18n: "i18n", active: "active", disabled: "disabled", expanded: "expanded", groupItem: "groupItem", hasFocus: "hasFocus", focusElement: "focusElement", itemText: "itemText" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationItemDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'cds-navigation-item' }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { i18n: [{
                type: Input
            }], active: [{
                type: Input
            }], disabled: [{
                type: Input
            }], expanded: [{
                type: Input
            }], groupItem: [{
                type: Input
            }], hasFocus: [{
                type: Input
            }], focusElement: [{
                type: Input
            }], itemText: [{
                type: Input
            }] } });
