// @cds/angular/esm2020/navigation/cds-navigation-start.directive.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class CdsNavigationStartDirective {
    constructor(elementRef) {
        this.element = elementRef.nativeElement;
    }
    get i18n() {
        return this.element.i18n;
    }
    set i18n(value) {
        this.element.i18n = value;
    }
    ;
    get expandedRoot() {
        return this.element.expandedRoot;
    }
    set expandedRoot(value) {
        this.element.expandedRoot = value === '' ? true : value;
    }
    ;
    get hasFocus() {
        return this.element.hasFocus;
    }
    set hasFocus(value) {
        this.element.hasFocus = value;
    }
    ;
    get isGroupStart() {
        return this.element.isGroupStart;
    }
    set isGroupStart(value) {
        this.element.isGroupStart = value === '' ? true : value;
    }
    ;
    get navigationGroupId() {
        return this.element.navigationGroupId;
    }
    set navigationGroupId(value) {
        this.element.navigationGroupId = value;
    }
    ;
    get itemText() {
        return this.element.itemText;
    }
    set itemText(value) {
        this.element.itemText = value;
    }
    ;
}
CdsNavigationStartDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationStartDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
CdsNavigationStartDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: CdsNavigationStartDirective, selector: "cds-navigation-start", inputs: { i18n: "i18n", expandedRoot: "expandedRoot", hasFocus: "hasFocus", isGroupStart: "isGroupStart", navigationGroupId: "navigationGroupId", itemText: "itemText" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationStartDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'cds-navigation-start' }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { i18n: [{
                type: Input
            }], expandedRoot: [{
                type: Input
            }], hasFocus: [{
                type: Input
            }], isGroupStart: [{
                type: Input
            }], navigationGroupId: [{
                type: Input
            }], itemText: [{
                type: Input
            }] } });
