// @cds/angular/esm2020/navigation/navigation.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/navigation/register.js';
import { CdsNavigationDirective } from './cds-navigation.directive';
import { CdsNavigationGroupDirective } from './cds-navigation-group.directive';
import { CdsNavigationStartDirective } from './cds-navigation-start.directive';
import { CdsNavigationItemDirective } from './cds-navigation-item.directive';
import * as i0 from "@angular/core";
export class CdsNavigationModule {
}
CdsNavigationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsNavigationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationModule, declarations: [CdsNavigationDirective,
        CdsNavigationGroupDirective,
        CdsNavigationStartDirective,
        CdsNavigationItemDirective], imports: [CommonModule], exports: [CdsNavigationDirective,
        CdsNavigationGroupDirective,
        CdsNavigationStartDirective,
        CdsNavigationItemDirective] });
CdsNavigationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsNavigationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsNavigationDirective,
                        CdsNavigationGroupDirective,
                        CdsNavigationStartDirective,
                        CdsNavigationItemDirective,
                    ],
                    exports: [
                        CdsNavigationDirective,
                        CdsNavigationGroupDirective,
                        CdsNavigationStartDirective,
                        CdsNavigationItemDirective,
                    ],
                }]
        }] });
