// @cds/angular/esm2020/card/card.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/card/register.js';
import { CdsCardDirective } from './cds-card.directive';
import * as i0 from "@angular/core";
export class CdsCardModule {
}
CdsCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCardModule, declarations: [CdsCardDirective], imports: [CommonModule], exports: [CdsCardDirective] });
CdsCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCardModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsCardDirective,
                    ],
                    exports: [
                        CdsCardDirective,
                    ],
                }]
        }] });
