// @cds/angular/esm2020/breadcrumb/breadcrumb.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/breadcrumb/register.js';
import { CdsBreadcrumbDirective } from './cds-breadcrumb.directive';
import * as i0 from "@angular/core";
export class CdsBreadcrumbModule {
}
CdsBreadcrumbModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBreadcrumbModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsBreadcrumbModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBreadcrumbModule, declarations: [CdsBreadcrumbDirective], imports: [CommonModule], exports: [CdsBreadcrumbDirective] });
CdsBreadcrumbModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBreadcrumbModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBreadcrumbModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsBreadcrumbDirective,
                    ],
                    exports: [
                        CdsBreadcrumbDirective,
                    ],
                }]
        }] });
