// @cds/angular/esm2020/radio/radio.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/radio/register.js';
import { CdsRadioDirective } from './cds-radio.directive';
import { CdsRadioGroupDirective } from './cds-radio-group.directive';
import * as i0 from "@angular/core";
export class CdsRadioModule {
}
CdsRadioModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRadioModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsRadioModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRadioModule, declarations: [CdsRadioDirective,
        CdsRadioGroupDirective], imports: [CommonModule], exports: [CdsRadioDirective,
        CdsRadioGroupDirective] });
CdsRadioModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRadioModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsRadioModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsRadioDirective,
                        CdsRadioGroupDirective,
                    ],
                    exports: [
                        CdsRadioDirective,
                        CdsRadioGroupDirective,
                    ],
                }]
        }] });
