// @cds/angular/esm2020/input/input.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/input/register.js';
import { CdsInputDirective } from './cds-input.directive';
import { CdsInputGroupDirective } from './cds-input-group.directive';
import * as i0 from "@angular/core";
export class CdsInputModule {
}
CdsInputModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInputModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInputModule, declarations: [CdsInputDirective,
        CdsInputGroupDirective], imports: [CommonModule], exports: [CdsInputDirective,
        CdsInputGroupDirective] });
CdsInputModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInputModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInputModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInputDirective,
                        CdsInputGroupDirective,
                    ],
                    exports: [
                        CdsInputDirective,
                        CdsInputGroupDirective,
                    ],
                }]
        }] });
