// @cds/angular/fesm2020/internal-components-panel.mjs
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import '@cds/core/internal-components/panel/register.js';

/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
class CdsInternalPanelDirective {
    constructor(elementRef) {
        this.element = elementRef.nativeElement;
    }
}
CdsInternalPanelDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalPanelDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
CdsInternalPanelDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: CdsInternalPanelDirective, selector: "cds-internal-panel", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalPanelDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'cds-internal-panel' }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; } });

/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
class CdsInternalComponentsPanelModule {
}
CdsInternalComponentsPanelModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInternalComponentsPanelModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, declarations: [CdsInternalPanelDirective], imports: [CommonModule], exports: [CdsInternalPanelDirective] });
CdsInternalComponentsPanelModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInternalPanelDirective,
                    ],
                    exports: [
                        CdsInternalPanelDirective,
                    ],
                }]
        }] });

/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */

/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */

/**
 * Generated bundle index. Do not edit.
 */

export { CdsInternalComponentsPanelModule, CdsInternalPanelDirective };
