// @cds/angular/esm2020/progress-circle/cds-progress-circle.directive.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class CdsProgressCircleDirective {
    constructor(elementRef) {
        this.element = elementRef.nativeElement;
    }
    get status() {
        return this.element.status;
    }
    set status(value) {
        this.element.status = value;
    }
    ;
    get inverse() {
        return this.element.inverse;
    }
    set inverse(value) {
        this.element.inverse = value === '' ? true : value;
    }
    ;
    get value() {
        return this.element.value;
    }
    set value(value) {
        this.element.value = value;
    }
    ;
    get line() {
        return this.element.line;
    }
    set line(value) {
        this.element.line = value;
    }
    ;
    get size() {
        return this.element.size;
    }
    set size(value) {
        this.element.size = value;
    }
    ;
    get i18n() {
        return this.element.i18n;
    }
    set i18n(value) {
        this.element.i18n = value;
    }
    ;
}
CdsProgressCircleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsProgressCircleDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
CdsProgressCircleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: CdsProgressCircleDirective, selector: "cds-progress-circle", inputs: { status: "status", inverse: "inverse", value: "value", line: "line", size: "size", i18n: "i18n" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsProgressCircleDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'cds-progress-circle' }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { status: [{
                type: Input
            }], inverse: [{
                type: Input
            }], value: [{
                type: Input
            }], line: [{
                type: Input
            }], size: [{
                type: Input
            }], i18n: [{
                type: Input
            }] } });
