// @cds/angular/esm2020/progress-circle/progress-circle.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/progress-circle/register.js';
import { CdsProgressCircleDirective } from './cds-progress-circle.directive';
import * as i0 from "@angular/core";
export class CdsProgressCircleModule {
}
CdsProgressCircleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsProgressCircleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsProgressCircleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsProgressCircleModule, declarations: [CdsProgressCircleDirective], imports: [CommonModule], exports: [CdsProgressCircleDirective] });
CdsProgressCircleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsProgressCircleModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsProgressCircleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsProgressCircleDirective,
                    ],
                    exports: [
                        CdsProgressCircleDirective,
                    ],
                }]
        }] });
