// @cds/angular/esm2020/time/time.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/time/register.js';
import { CdsTimeDirective } from './cds-time.directive';
import * as i0 from "@angular/core";
export class CdsTimeModule {
}
CdsTimeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTimeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsTimeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTimeModule, declarations: [CdsTimeDirective], imports: [CommonModule], exports: [CdsTimeDirective] });
CdsTimeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTimeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTimeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsTimeDirective,
                    ],
                    exports: [
                        CdsTimeDirective,
                    ],
                }]
        }] });
