// @cds/angular/esm2020/tree-view/tree-view.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/tree-view/register.js';
import { CdsTreeDirective } from './cds-tree.directive';
import { CdsTreeItemDirective } from './cds-tree-item.directive';
import * as i0 from "@angular/core";
export class CdsTreeViewModule {
}
CdsTreeViewModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTreeViewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsTreeViewModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTreeViewModule, declarations: [CdsTreeDirective,
        CdsTreeItemDirective], imports: [CommonModule], exports: [CdsTreeDirective,
        CdsTreeItemDirective] });
CdsTreeViewModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTreeViewModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTreeViewModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsTreeDirective,
                        CdsTreeItemDirective,
                    ],
                    exports: [
                        CdsTreeDirective,
                        CdsTreeItemDirective,
                    ],
                }]
        }] });
