// @cds/angular/esm2020/date/date.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/date/register.js';
import { CdsDateDirective } from './cds-date.directive';
import * as i0 from "@angular/core";
export class CdsDateModule {
}
CdsDateModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsDateModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDateModule, declarations: [CdsDateDirective], imports: [CommonModule], exports: [CdsDateDirective] });
CdsDateModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDateModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsDateModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsDateDirective,
                    ],
                    exports: [
                        CdsDateDirective,
                    ],
                }]
        }] });
