// @cds/angular/esm2020/search/search.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/search/register.js';
import { CdsSearchDirective } from './cds-search.directive';
import * as i0 from "@angular/core";
export class CdsSearchModule {
}
CdsSearchModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSearchModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsSearchModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSearchModule, declarations: [CdsSearchDirective], imports: [CommonModule], exports: [CdsSearchDirective] });
CdsSearchModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSearchModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSearchModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsSearchDirective,
                    ],
                    exports: [
                        CdsSearchDirective,
                    ],
                }]
        }] });
