// @cds/angular/esm2020/file/file.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/file/register.js';
import { CdsFileDirective } from './cds-file.directive';
import * as i0 from "@angular/core";
export class CdsFileModule {
}
CdsFileModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFileModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsFileModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFileModule, declarations: [CdsFileDirective], imports: [CommonModule], exports: [CdsFileDirective] });
CdsFileModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFileModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFileModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsFileDirective,
                    ],
                    exports: [
                        CdsFileDirective,
                    ],
                }]
        }] });
