// @cds/angular/esm2020/internal-components/panel/internal-components-panel.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/internal-components/panel/register.js';
import { CdsInternalPanelDirective } from './cds-internal-panel.directive';
import * as i0 from "@angular/core";
export class CdsInternalComponentsPanelModule {
}
CdsInternalComponentsPanelModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInternalComponentsPanelModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, declarations: [CdsInternalPanelDirective], imports: [CommonModule], exports: [CdsInternalPanelDirective] });
CdsInternalComponentsPanelModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPanelModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInternalPanelDirective,
                    ],
                    exports: [
                        CdsInternalPanelDirective,
                    ],
                }]
        }] });
