// @cds/angular/esm2020/select/select.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/select/register.js';
import { CdsSelectDirective } from './cds-select.directive';
import * as i0 from "@angular/core";
export class CdsSelectModule {
}
CdsSelectModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsSelectModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectModule, declarations: [CdsSelectDirective], imports: [CommonModule], exports: [CdsSelectDirective] });
CdsSelectModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsSelectDirective,
                    ],
                    exports: [
                        CdsSelectDirective,
                    ],
                }]
        }] });
