// @cds/angular/esm2020/internal-components/split-handle/internal-components-split-handle.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/internal-components/split-handle/register.js';
import { CdsInternalSplitHandleDirective } from './cds-internal-split-handle.directive';
import * as i0 from "@angular/core";
export class CdsInternalComponentsSplitHandleModule {
}
CdsInternalComponentsSplitHandleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsSplitHandleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInternalComponentsSplitHandleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsSplitHandleModule, declarations: [CdsInternalSplitHandleDirective], imports: [CommonModule], exports: [CdsInternalSplitHandleDirective] });
CdsInternalComponentsSplitHandleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsSplitHandleModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsSplitHandleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInternalSplitHandleDirective,
                    ],
                    exports: [
                        CdsInternalSplitHandleDirective,
                    ],
                }]
        }] });
