// @cds/angular/esm2020/textarea/textarea.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/textarea/register.js';
import { CdsTextareaDirective } from './cds-textarea.directive';
import * as i0 from "@angular/core";
export class CdsTextareaModule {
}
CdsTextareaModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTextareaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsTextareaModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTextareaModule, declarations: [CdsTextareaDirective], imports: [CommonModule], exports: [CdsTextareaDirective] });
CdsTextareaModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTextareaModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTextareaModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsTextareaDirective,
                    ],
                    exports: [
                        CdsTextareaDirective,
                    ],
                }]
        }] });
