// @cds/angular/esm2020/accordion/accordion.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/accordion/register.js';
import { CdsAccordionDirective } from './cds-accordion.directive';
import { CdsAccordionPanelDirective } from './cds-accordion-panel.directive';
import { CdsAccordionContentDirective } from './cds-accordion-content.directive';
import { CdsAccordionHeaderDirective } from './cds-accordion-header.directive';
import * as i0 from "@angular/core";
export class CdsAccordionModule {
}
CdsAccordionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAccordionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsAccordionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAccordionModule, declarations: [CdsAccordionDirective,
        CdsAccordionPanelDirective,
        CdsAccordionContentDirective,
        CdsAccordionHeaderDirective], imports: [CommonModule], exports: [CdsAccordionDirective,
        CdsAccordionPanelDirective,
        CdsAccordionContentDirective,
        CdsAccordionHeaderDirective] });
CdsAccordionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAccordionModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAccordionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsAccordionDirective,
                        CdsAccordionPanelDirective,
                        CdsAccordionContentDirective,
                        CdsAccordionHeaderDirective,
                    ],
                    exports: [
                        CdsAccordionDirective,
                        CdsAccordionPanelDirective,
                        CdsAccordionContentDirective,
                        CdsAccordionHeaderDirective,
                    ],
                }]
        }] });
