// @cds/angular/esm2020/badge/badge.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/badge/register.js';
import { CdsBadgeDirective } from './cds-badge.directive';
import * as i0 from "@angular/core";
export class CdsBadgeModule {
}
CdsBadgeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsBadgeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBadgeModule, declarations: [CdsBadgeDirective], imports: [CommonModule], exports: [CdsBadgeDirective] });
CdsBadgeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBadgeModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsBadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsBadgeDirective,
                    ],
                    exports: [
                        CdsBadgeDirective,
                    ],
                }]
        }] });
