// @cds/angular/esm2020/button/button.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/button/register.js';
import { CdsButtonDirective } from './cds-button.directive';
import { CdsIconButtonDirective } from './cds-icon-button.directive';
import * as i0 from "@angular/core";
export class CdsButtonModule {
}
CdsButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonModule, declarations: [CdsButtonDirective,
        CdsIconButtonDirective], imports: [CommonModule], exports: [CdsButtonDirective,
        CdsIconButtonDirective] });
CdsButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsButtonDirective,
                        CdsIconButtonDirective,
                    ],
                    exports: [
                        CdsButtonDirective,
                        CdsIconButtonDirective,
                    ],
                }]
        }] });
