// @cds/angular/esm2020/pagination/pagination.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/pagination/register.js';
import { CdsPaginationDirective } from './cds-pagination.directive';
import { CdsPaginationButtonDirective } from './cds-pagination-button.directive';
import * as i0 from "@angular/core";
export class CdsPaginationModule {
}
CdsPaginationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsPaginationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsPaginationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsPaginationModule, declarations: [CdsPaginationDirective,
        CdsPaginationButtonDirective], imports: [CommonModule], exports: [CdsPaginationDirective,
        CdsPaginationButtonDirective] });
CdsPaginationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsPaginationModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsPaginationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsPaginationDirective,
                        CdsPaginationButtonDirective,
                    ],
                    exports: [
                        CdsPaginationDirective,
                        CdsPaginationButtonDirective,
                    ],
                }]
        }] });
