// @cds/angular/esm2020/checkbox/checkbox.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/checkbox/register.js';
import { CdsCheckboxDirective } from './cds-checkbox.directive';
import { CdsCheckboxGroupDirective } from './cds-checkbox-group.directive';
import * as i0 from "@angular/core";
export class CdsCheckboxModule {
}
CdsCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCheckboxModule, declarations: [CdsCheckboxDirective,
        CdsCheckboxGroupDirective], imports: [CommonModule], exports: [CdsCheckboxDirective,
        CdsCheckboxGroupDirective] });
CdsCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCheckboxModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsCheckboxDirective,
                        CdsCheckboxGroupDirective,
                    ],
                    exports: [
                        CdsCheckboxDirective,
                        CdsCheckboxGroupDirective,
                    ],
                }]
        }] });
