// @cds/angular/esm2020/modal/modal.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/modal/register.js';
import { CdsModalHeaderDirective } from './cds-modal-header.directive';
import { CdsModalHeaderActionsDirective } from './cds-modal-header-actions.directive';
import { CdsModalContentDirective } from './cds-modal-content.directive';
import { CdsModalActionsDirective } from './cds-modal-actions.directive';
import { CdsModalDirective } from './cds-modal.directive';
import * as i0 from "@angular/core";
export class CdsModalModule {
}
CdsModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsModalModule, declarations: [CdsModalHeaderDirective,
        CdsModalHeaderActionsDirective,
        CdsModalContentDirective,
        CdsModalActionsDirective,
        CdsModalDirective], imports: [CommonModule], exports: [CdsModalHeaderDirective,
        CdsModalHeaderActionsDirective,
        CdsModalContentDirective,
        CdsModalActionsDirective,
        CdsModalDirective] });
CdsModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsModalModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsModalHeaderDirective,
                        CdsModalHeaderActionsDirective,
                        CdsModalContentDirective,
                        CdsModalActionsDirective,
                        CdsModalDirective,
                    ],
                    exports: [
                        CdsModalHeaderDirective,
                        CdsModalHeaderActionsDirective,
                        CdsModalContentDirective,
                        CdsModalActionsDirective,
                        CdsModalDirective,
                    ],
                }]
        }] });
