// @cds/angular/esm2020/selection-panels/radio/selection-panels-radio.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/selection-panels/radio/register.js';
import { CdsRadioPanelDirective } from './cds-radio-panel.directive';
import * as i0 from "@angular/core";
export class CdsSelectionPanelsRadioModule {
}
CdsSelectionPanelsRadioModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsRadioModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsSelectionPanelsRadioModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsRadioModule, declarations: [CdsRadioPanelDirective], imports: [CommonModule], exports: [CdsRadioPanelDirective] });
CdsSelectionPanelsRadioModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsRadioModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsSelectionPanelsRadioModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsRadioPanelDirective,
                    ],
                    exports: [
                        CdsRadioPanelDirective,
                    ],
                }]
        }] });
