// @cds/angular/esm2020/internal-components/popup/internal-components-popup.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/internal-components/popup/register.js';
import { CdsInternalPopupDirective } from './cds-internal-popup.directive';
import { CdsInternalPointerDirective } from './cds-internal-pointer.directive';
import * as i0 from "@angular/core";
export class CdsInternalComponentsPopupModule {
}
CdsInternalComponentsPopupModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPopupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsInternalComponentsPopupModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPopupModule, declarations: [CdsInternalPopupDirective,
        CdsInternalPointerDirective], imports: [CommonModule], exports: [CdsInternalPopupDirective,
        CdsInternalPointerDirective] });
CdsInternalComponentsPopupModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPopupModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsInternalComponentsPopupModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsInternalPopupDirective,
                        CdsInternalPointerDirective,
                    ],
                    exports: [
                        CdsInternalPopupDirective,
                        CdsInternalPointerDirective,
                    ],
                }]
        }] });
