// @cds/angular/esm2020/button-sort/button-sort.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/button-sort/register.js';
import { CdsButtonSortDirective } from './cds-button-sort.directive';
import * as i0 from "@angular/core";
export class CdsButtonSortModule {
}
CdsButtonSortModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonSortModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsButtonSortModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonSortModule, declarations: [CdsButtonSortDirective], imports: [CommonModule], exports: [CdsButtonSortDirective] });
CdsButtonSortModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonSortModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonSortModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsButtonSortDirective,
                    ],
                    exports: [
                        CdsButtonSortDirective,
                    ],
                }]
        }] });
