// @cds/angular/esm2020/toggle/toggle.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/toggle/register.js';
import { CdsToggleDirective } from './cds-toggle.directive';
import { CdsToggleGroupDirective } from './cds-toggle-group.directive';
import * as i0 from "@angular/core";
export class CdsToggleModule {
}
CdsToggleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsToggleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsToggleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsToggleModule, declarations: [CdsToggleDirective,
        CdsToggleGroupDirective], imports: [CommonModule], exports: [CdsToggleDirective,
        CdsToggleGroupDirective] });
CdsToggleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsToggleModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsToggleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsToggleDirective,
                        CdsToggleGroupDirective,
                    ],
                    exports: [
                        CdsToggleDirective,
                        CdsToggleGroupDirective,
                    ],
                }]
        }] });
