// @cds/angular/esm2020/tag/tag.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/tag/register.js';
import { CdsTagDirective } from './cds-tag.directive';
import * as i0 from "@angular/core";
export class CdsTagModule {
}
CdsTagModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTagModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsTagModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTagModule, declarations: [CdsTagDirective], imports: [CommonModule], exports: [CdsTagDirective] });
CdsTagModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTagModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsTagModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsTagDirective,
                    ],
                    exports: [
                        CdsTagDirective,
                    ],
                }]
        }] });
