// @cds/angular/esm2020/button-action/button-action.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/button-action/register.js';
import { CdsButtonActionDirective } from './cds-button-action.directive';
import * as i0 from "@angular/core";
export class CdsButtonActionModule {
}
CdsButtonActionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonActionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsButtonActionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonActionModule, declarations: [CdsButtonActionDirective], imports: [CommonModule], exports: [CdsButtonActionDirective] });
CdsButtonActionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonActionModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonActionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsButtonActionDirective,
                    ],
                    exports: [
                        CdsButtonActionDirective,
                    ],
                }]
        }] });
