// @cds/angular/esm2020/button-inline/button-inline.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/button-inline/register.js';
import { CdsButtonInlineDirective } from './cds-button-inline.directive';
import * as i0 from "@angular/core";
export class CdsButtonInlineModule {
}
CdsButtonInlineModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonInlineModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsButtonInlineModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonInlineModule, declarations: [CdsButtonInlineDirective], imports: [CommonModule], exports: [CdsButtonInlineDirective] });
CdsButtonInlineModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonInlineModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonInlineModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsButtonInlineDirective,
                    ],
                    exports: [
                        CdsButtonInlineDirective,
                    ],
                }]
        }] });
