// @cds/angular/esm2020/forms/forms.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/forms/register.js';
import { CdsControlMessageDirective } from './cds-control-message.directive';
import { CdsControlActionDirective } from './cds-control-action.directive';
import { CdsInternalControlGroupDirective } from './cds-internal-control-group.directive';
import { CdsControlDirective } from './cds-control.directive';
import { CdsFormGroupDirective } from './cds-form-group.directive';
import { CdsInternalControlInlineDirective } from './cds-internal-control-inline.directive';
import { CdsInternalControlLabelDirective } from './cds-internal-control-label.directive';
import * as i0 from "@angular/core";
export class CdsFormsModule {
}
CdsFormsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFormsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsFormsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFormsModule, declarations: [CdsControlMessageDirective,
        CdsControlActionDirective,
        CdsInternalControlGroupDirective,
        CdsControlDirective,
        CdsFormGroupDirective,
        CdsInternalControlInlineDirective,
        CdsInternalControlLabelDirective], imports: [CommonModule], exports: [CdsControlMessageDirective,
        CdsControlActionDirective,
        CdsInternalControlGroupDirective,
        CdsControlDirective,
        CdsFormGroupDirective,
        CdsInternalControlInlineDirective,
        CdsInternalControlLabelDirective] });
CdsFormsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFormsModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsFormsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsControlMessageDirective,
                        CdsControlActionDirective,
                        CdsInternalControlGroupDirective,
                        CdsControlDirective,
                        CdsFormGroupDirective,
                        CdsInternalControlInlineDirective,
                        CdsInternalControlLabelDirective,
                    ],
                    exports: [
                        CdsControlMessageDirective,
                        CdsControlActionDirective,
                        CdsInternalControlGroupDirective,
                        CdsControlDirective,
                        CdsFormGroupDirective,
                        CdsInternalControlInlineDirective,
                        CdsInternalControlLabelDirective,
                    ],
                }]
        }] });
