// @cds/angular/esm2020/button-expand/button-expand.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/button-expand/register.js';
import { CdsButtonExpandDirective } from './cds-button-expand.directive';
import * as i0 from "@angular/core";
export class CdsButtonExpandModule {
}
CdsButtonExpandModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonExpandModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsButtonExpandModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonExpandModule, declarations: [CdsButtonExpandDirective], imports: [CommonModule], exports: [CdsButtonExpandDirective] });
CdsButtonExpandModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonExpandModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsButtonExpandModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsButtonExpandDirective,
                    ],
                    exports: [
                        CdsButtonExpandDirective,
                    ],
                }]
        }] });
