// @cds/angular/esm2020/alert/alert.module.mjs
/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import '@cds/core/alert/register.js';
import { CdsAlertActionsDirective } from './cds-alert-actions.directive';
import { CdsAlertDirective } from './cds-alert.directive';
import { CdsAlertGroupDirective } from './cds-alert-group.directive';
import * as i0 from "@angular/core";
export class CdsAlertModule {
}
CdsAlertModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAlertModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CdsAlertModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAlertModule, declarations: [CdsAlertActionsDirective,
        CdsAlertDirective,
        CdsAlertGroupDirective], imports: [CommonModule], exports: [CdsAlertActionsDirective,
        CdsAlertDirective,
        CdsAlertGroupDirective] });
CdsAlertModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAlertModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CdsAlertModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        CdsAlertActionsDirective,
                        CdsAlertDirective,
                        CdsAlertGroupDirective,
                    ],
                    exports: [
                        CdsAlertActionsDirective,
                        CdsAlertDirective,
                        CdsAlertGroupDirective,
                    ],
                }]
        }] });
